ARCHITECTURE.md

```diff
@@ -0,0 +1,158 @@
+    
+      // The type of data carried by attributes
+      ValueType value_type = 3;
+    }
+
+    enum ValueType {
+        // Invalid, default value.
+        VALUE_TYPE_UNSPECIFIED = 0;
+    
+        // An undiscriminated variable-length string.
+        STRING = 1;
+    
+        // An undiscriminated 64-bit signed integer.
+        INT64 = 2;
+    
+        // An undiscriminated 64-bit floating-point value.
+        DOUBLE = 3;
+    
+        // An undiscriminated boolean value.
+        BOOL = 4;
+    
+        // A point in time.
+        TIMESTAMP = 5;
+    
+        // An IP address.
+        IP_ADDRESS = 6;
+    
+        // An email address.
+        EMAIL_ADDRESS = 7;
+    
+        // A URI.
+        URI = 8;
+    
+        // A DNS name.
+        DNS_NAME = 9;
+    
+        // A span between two points in time.
+        DURATION = 10;
+    
+        // A map string -> string, typically used by headers.
+        STRING_MAP = 11;
+    }
+
+When an attribute is used in Istio, its name is given along with a value. The value must be of the type declared by the corresponding descriptor. This
+type checking makes it possible for the Istio system to statically detect or prevent many configuration errors.
+
+## Configuration
+
+**TBD**
+
+## The proxy
+
+**TBD**
+
+## The mixer
+
+The mixer provides the fundamental abstractions common to most multi-tenant SaaS products. Services built atop the mixer are insulated from platform-specific
+details enabling them to deploy on multiple platforms (different clouds, on-prem, private), while also being able to serve customers coming from different
+platforms in a first-class way. The mixer sits at the bottom of an Istio deployment, between the service proper and the backends necessary to support it. It 
+is invoked by both the Istio proxy as well as the service proper.
+
+The mixer provides three core features:
+
+- **Precondition Checking**. Enables callers to verify a number of preconditions before responding to an incoming request from a service consumer. 
+Preconditions can include whether the service consumer is properly authenticated, is on the service's whitelist, has the right access privilege, and more.
+
+- **Telemetry Reporting**. Enables services to produce logging, monitoring, tracing and billing streams intended for the service producer itself as well as 
+for its consumers.
+
+- **Quota Management**. Enables services to allocate and free quota on a number of dimensions, Quotas are used as a relatively simple resource management 
+tool to provide some fairness between service consumers when contending for limited service resources.
+
+### Adapters
+
+Adapters are binary-level plugins to the mixer which make it possible to customize the mixer’s behavior. For example, adapters allow the mixer to interface to
+different underlying platforms (distinct clouds, on-prem systems, etc). Different deployments of the mixer feature different sets of adapters, determining
```

Comment:
I would flesh out the example here a little more to include monitoring, logging & ACL systems. 

The term 'underlying platforms' is a little unsatisfying when in reality the mixer is creating a platform by pulling together a number of different concerns